contracts/abstract/IdBinarySearch.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.6;
+
+/**
+ * @title  PoolTogether V4 IdBinarySearch
+ * @author PoolTogether Inc Team
+ * @notice IdBinarySearch uses binary search to find a parent contract struct with the drawId parameter
+ * @dev    The implementing contract must provider access to a struct (i.e. PrizeTier) list with is both
+ *         sorted and indexed by the drawId field for binary search to work.
+ */
+abstract contract IdBinarySearch {
+    /**
+     * @notice Get newest index in array
+     */
+    function getNewestIndex() internal view virtual returns (uint32);
+
+    /**
+     * @notice Get Draw ID for using an index position
+     * @param index uint256 - Index of element in array
+     */
+    function getIdForIndex(uint256 index) internal view virtual returns (uint32);
+
+    function _binarySearch(uint32 _drawId) internal view returns (uint32) {
```

Comment:
This function would instead take a `uint32[] storage` field as first arg.  You won't need the `getIdForIndex` or the `getNewestIndex` because both would be readily available in the array itself.

Reply:
Fixed.